src/vs/workbench/api/common/shared/webview.ts

```diff
@@ -55,9 +55,15 @@ export function asWebviewUri(
 		});
 	}
 
+	let authority = `${resource.scheme}+${resource.authority}.${webviewRootResourceAuthority}`;
+	if (resource.authority.includes('localhost')) {
+		const authorityUrl = new URL(`http://${resource.authority}`);
```

Comment:
To fix the layer errors, try using our `URI` class instead of the built-in `URL` class

Reply:
I think I tried that before but the `URI` class doesn't have a port (though it does have it in the `authority` but then I'd have to split it on the `:`, right?) Unless I leave the port out entirely?

Reply:
@mjbvz The URI puts the `authority` section before the `vscode-resource.vscode-webview.net`. So the url becomes `localhost:8080.vscode-resource.vscode-webview.net` which is an invalid port.

Ideally, the `URI` package could parse the port like `URL`.